deployer/client/client.go

```diff
@@ -0,0 +1,81 @@
+package client
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+
+	"github.com/flynn/flynn/deployer/types"
+	"github.com/flynn/flynn/discoverd/client"
+	"github.com/flynn/flynn/discoverd/client/dialer"
+	"github.com/flynn/flynn/pkg/httpclient"
+	"github.com/flynn/flynn/pkg/sse"
+)
+
+// ErrNotFound is returned when no job was found.
+var ErrNotFound = errors.New("deployer: not found")
+
+type Client struct {
+	*httpclient.Client
+}
+
+// New uses the default discoverd client and returns a client.
+func New() (*Client, error) {
+	if err := discoverd.Connect(""); err != nil {
+		return nil, err
+	}
+	dialer := dialer.New(discoverd.DefaultClient, nil)
+	c := &httpclient.Client{
+		Dial:        dialer.Dial,
+		DialClose:   dialer,
+		ErrPrefix:   "deployer",
+		ErrNotFound: ErrNotFound,
+		URL:         "http://flynn-deployer",
+	}
+	c.HTTP = &http.Client{Transport: &http.Transport{Dial: c.Dial}}
+	return &Client{Client: c}, nil
+}
+
+func (c *Client) CreateDeployment(deployment *deployer.Deployment) error {
+	return c.Post("/deployments", deployment, deployment)
+}
+
+// DeploymentEventStream is a wrapper around an Events channel, allowing us to close
+// the stream.
+type DeploymentEventStream struct {
+	Events chan *deployer.DeploymentEvent
+	body   io.ReadCloser
+}
+
+// Close closes the underlying stream.
+func (s *DeploymentEventStream) Close() {
+	s.body.Close()
+}
+
+// StreamDeploymentEvents returns a DeploymentEventStream for an app.
+func (c *Client) StreamDeploymentEvents(deploymentID string, lastID int64) (*DeploymentEventStream, error) {
```

Comment:
This should use the [Stream interface](https://github.com/flynn/flynn/blob/master/pkg/stream/stream.go).